src/Database/PG/Query/Pool.hs

```diff
@@ -65,23 +81,46 @@ data ConnParams
 defaultConnParams :: ConnParams
 defaultConnParams = ConnParams 1 20 60 True Nothing
 
+data ConnectionHooks
+  = ConnectionHooks
+  { connectionHookPostCreate  :: PGConn -> IO ()
+    -- ^ post connection init hook. Is the return type () sensible?
```

Comment:
> Is the return type () sensible?

What is your thinking here? seems sensible to me

Reply:
Sorry, I should remove the comment. Intially I wrote it down (thinking should the hook function perform some side-effect or should return something), but I think this makes sense.